Case insensitive string comparison beaks several long integration tests

### Search before asking

- [X] I have searched the EvaDB [issues](https://github.com/georgia-tech-db/eva/issues) and found no similar bug report.


### Bug

[Circle Ci](https://app.circleci.com/pipelines/github/georgia-tech-db/evadb/5328/workflows/39e5255f-88cc-4c57-83b7-d3da854fd361/jobs/38343) if the test does not make sense, we can also modify or clean up the test. We should not compare string with None. 

```
=========================== short test summary info ============================
FAILED test/integration_tests/long/test_function_executor.py::FunctionExecutorTest::test_create_function_with_decorators - evadb.executor.executor_utils.ExecutorError: 'NoneType' object has no attribute 'lower'
FAILED test/integration_tests/long/test_single_document_similarity.py::SingleDocumentSimilarityTests::test_single_pdf_should_work - evadb.executor.executor_utils.ExecutorError: 'NoneType' object has no attribute 'lower'
FAILED test/integration_tests/long/test_text_filtering.py::TextFilteringTests::test_text_filter - evadb.executor.executor_utils.ExecutorError: 'NoneType' object has no attribute 'lower'
FAILED test/integration_tests/long/interfaces/relational/test_relational_api.py::RelationalAPI::test_pdf_similarity_search - evadb.executor.executor_utils.ExecutorError: 'NoneType' object has no attribute 'lower'
= 4 failed, 122 passed, 32 skipped, 6 deselected, 12 warnings in 413.23s (0:06:53) =
LONG INTEGRATION TEST CODE: --|1|-- FAILURE
```

### Environment

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

--- a/evadb/utils/generic_utils.py
+++ b/evadb/utils/generic_utils.py
@@ -535,4 +535,8 @@ def string_comparison_case_insensitive(string_1, string_2) -> bool:
         True/False (bool): Returns True if the strings are same, false otherwise
     """
 
+    # Does not make sense in case of null strings
+    if string_1 is None or string_2 is None:
+        return False
+
     return string_1.lower() == string_2.lower()

--- a/test/unit_tests/binder/test_statement_binder.py
+++ b/test/unit_tests/binder/test_statement_binder.py
@@ -463,8 +463,8 @@ class StatementBinderTests(unittest.TestCase):
         test_one_string_null = string_comparison_case_insensitive(None, "HuggingFace")
         test_both_strings_null = string_comparison_case_insensitive(None, None)
 
-        assert test_string_exact_match == True
-        assert test_string_case_insensitive_match == True
-        assert test_string_no_match == False
-        assert test_one_string_null == False
-        assert test_both_strings_null == False
+        assert test_string_exact_match is True
+        assert test_string_case_insensitive_match is True
+        assert test_string_no_match is False
+        assert test_one_string_null is False
+        assert test_both_strings_null is False

--- a/test/unit_tests/binder/test_statement_binder.py
+++ b/test/unit_tests/binder/test_statement_binder.py
@@ -24,6 +24,7 @@ from evadb.catalog.sql_config import IDENTIFIER_COLUMN
 from evadb.expression.tuple_value_expression import TupleValueExpression
 from evadb.parser.alias import Alias
 from evadb.parser.create_statement import ColumnDefinition
+from evadb.utils.generic_utils import string_comparison_case_insensitive
 
 
 def assert_not_called_with(self, *args, **kwargs):
@@ -443,3 +444,27 @@ class StatementBinderTests(unittest.TestCase):
             ]
             self.assertEqual(create_function_statement.inputs, expected_inputs)
             self.assertEqual(create_function_statement.outputs, expected_outputs)
+
+    def test_string_matching_case_insensitive(self):
+        """
+        A simple test for string_matching_case_insensitve in generic_utils
+        used by statement_binder
+        """
+
+        test_string_exact_match = string_comparison_case_insensitive(
+            "HuggingFace", "HuggingFace"
+        )
+        test_string_case_insensitive_match = string_comparison_case_insensitive(
+            "HuggingFace", "hugGingFaCe"
+        )
+        test_string_no_match = string_comparison_case_insensitive(
+            "HuggingFace", "HuggingFae"
+        )
+        test_one_string_null = string_comparison_case_insensitive(None, "HuggingFace")
+        test_both_strings_null = string_comparison_case_insensitive(None, None)
+
+        assert test_string_exact_match == True
+        assert test_string_case_insensitive_match == True
+        assert test_string_no_match == False
+        assert test_one_string_null == False
+        assert test_both_strings_null == False

